nox/sessions.py

```diff
@@ -417,15 +428,21 @@ def install(self, *args: str, **kwargs: Any) -> None:
 
         .. _pip: https://pip.readthedocs.org
         """
+        venv = self._runner.venv
+
         if not isinstance(
-            self._runner.venv, (CondaEnv, VirtualEnv, PassthroughEnv)
+            venv, (CondaEnv, VirtualEnv, PassthroughEnv)
         ):  # pragma: no cover
             raise ValueError(
                 "A session without a virtualenv can not install dependencies."
             )
         if not args:
             raise ValueError("At least one argument required to install().")
 
+        if self._runner.global_config.no_install and venv._reused:
+            logger.info("Venv exists: skipping installation, as --no-install is set.")
```

Comment:
I had added these warnings because I thought that an end-user may appreciate the heads-up that a non-default behaviour was occurring with the --no-install flags. However, I don't know that they're actually _necessary_, especially if the flags are well documented.

Let me remove them and see what we think 🙂